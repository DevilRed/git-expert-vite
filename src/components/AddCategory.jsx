import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('')
  const onInputChange = ({ target }) => {
    setInputValue(target.value)
  }
  const onSubmit = (event) => {
    event.preventDefault();
    setCategories(categories => [...categories, inputValue]);// using callback due conserv old cats
  }
  return (
    <form onSubmit={ onSubmit }>
      <input
        type="text"
        placeholder="Search gifs"
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}