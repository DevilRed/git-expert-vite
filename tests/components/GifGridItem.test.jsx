import { describe, it, expect } from 'vitest';
import { render } from "@testing-library/react";
import { GifGridItem } from '../../src/components/GifGridItem';

describe('GifGridItem', () => {
	it('should be match with snapshot', () => {
		const { asFragment } = render(<GifGridItem title={'some'} url={'someUrl'} />)
		expect(asFragment()).toMatchSnapshot();
	});
});