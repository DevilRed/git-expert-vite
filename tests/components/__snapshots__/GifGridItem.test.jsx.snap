// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifGridItem > should be match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <img
      alt="some"
      src="someUrl"
    />
    <p>
      some
    </p>
  </div>
</DocumentFragment>
`;
